### PR TITLE
soc/power9: implement istep 13.6

### PR DESCRIPTION
**Pull request by krystian-hebel**
Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

### CODE REVIEW

File: src/soc/ibm/power9/istep_13_6.c [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+static inline void p9_mem_startclocks_cplt_ctrl_action_function(chiplet_id_t id)
+{
+	// Drop partial good fences
+	/*
+	TP.TCMC01.MCSLOW.CPLT_CTRL1 (WO_CLEAR)                // 0x07000021
+	  [all]   0
+	  [3]     TC_VITL_REGION_FENCE =                  ~ATTR_PG[3]
+	  [4-14]  TC_REGION{1-3}_FENCE, UNUSED_{8-14}B =  ~ATTR_PG[4-14]
+	*/
+	write_scom_for_chiplet(id, 0x07000021, ~ATTR_PG & PPC_BITMASK(3,14));
```
Comment (miczyg1) [inline src/soc/ibm/power9/istep_13_6.c:34]: Register definitions in preprocessor along with bits

File: src/soc/ibm/power9/istep_13_6.c [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+	udelay(100);
+
+	// Poll aligned bit
+	/*
+	timeout(10*100us):
+	TP.TCMC01.MCSLOW.CPLT_STAT0                         // 0x07000100
+	if (([9] CC_CTRL_CHIPLET_IS_ALIGNED_DC) == 1) break
+	delay(100us)
+	*/
+	if (!wait_us(10 * 100, read_scom_for_chiplet(id, 0x07000100) & PPC_BIT(9)))
```
Comment (miczyg1) [inline src/soc/ibm/power9/istep_13_6.c:88]: I see the delay of 100000ns which is 100us, so why 10*100us?

File: src/soc/ibm/power9/istep_13_6.c [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+	                        | (~ATTR_PG & PPC_BITMASK(4,14)));
+
+	// Poll OPCG done bit to check for completeness
+	/*
+	timeout(10*100us):
+	TP.TCMC01.MCSLOW.CPLT_STAT0                         // 0x07000100
+	if (([8] CC_CTRL_OPCG_DONE_DC) == 1) break
+	delay(100us)
+	*/
+	if (!wait_us(10 * 100, read_scom_for_chiplet(id, 0x07000100) & PPC_BIT(8)))
```
Comment (miczyg1) [inline src/soc/ibm/power9/istep_13_6.c:143]: Same as above with timeout

File: src/soc/ibm/power9/istep_13_6.c [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+	  [18]  FENCE_EN =  0
+	*/
+	write_scom_for_chiplet(id, 0x070F0041, ~PPC_BIT(18));
+
+	/* }*/
+}
+
+static void p9_sbe_common_configure_chiplet_FIR(chiplet_id_t id)
+{
+	// reset pervasive FIR
```
Comment (miczyg1) [inline src/soc/ibm/power9/istep_13_6.c:193]: How do we know which l_unit_idx is this?

File: src/soc/ibm/power9/istep_13_6.c [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+	bool mcs_functional[MCS_PER_PROC] = {false, false};
+	chiplet_id_t mcs_ids[MCS_PER_PROC] = {MC01_CHIPLET_ID, MC23_CHIPLET_ID};
+
+	report_istep(13,6);
+
+	/* We already did it in 13.2. Maybe this should be done on a higher level? */
+	for (i = 0; i < CONFIG_DIMM_MAX; i++) {
+		if (dimms[i].dram_type == SPD_MEMORY_TYPE_DDR4_SDRAM &&
+		    dimms[i].dimm_type == SPD_DIMM_TYPE_RDIMM &&
+		    dimms[i].ecc_extension == true) {
```
Comment (miczyg1) [inline src/soc/ibm/power9/istep_13_6.c:259]: So the DIMMs must have ECC in order to work?

**Review by miczyg1 — CHANGES_REQUESTED** [review]

**Comment by macpijan** [comment]
Already integrated in: https://github.com/3mdeb/coreboot/pull/49

**Pull request closed without merging**